### DISCUSSION TITLE
CLI Run

### DISCUSSION THREAD

**Discussion by m-jabari** [jplag/JPlag#262] [Q&A] [answered]
Hey Guys,
Would you please, get me a minimum executable CLI command?

My CLI info:
ubuntu 20
*.c source code folders: **~/Desktop/fnFiles**
lang: ***.c**
root jplag git clone folder: **~/Desktop/jplag**
jar generated folder: **~/Desktop/jplag/jplag/target**
report folder: **~/Desktop/report**
my sample, which I'm calling it from **~/Desktop/jplag/jplag/target*:

> _java -jar jplag-3.0.0-SNAPSHOT-jar-with-dependencies.jar_ -l c_cpp -bc /home/ubuntu/Desktop/jplag/jplag/target  -v details -d Debug -S /home/ubuntu/Desktop/jplag/jplag/target -t 1 -s 0.7 -r ~/Desktop/report

Thank you

**Comment by tsaglam** [comment, marked as answer]
It seems like you are building your own jar from our current pre-release build. I would recommend you to use the latest [release](https://github.com/jplag/jplag/releases), as v3.0.0 is still very much work in progress. When using the legacy version of JPlag (2.12.1) call the jar without CLI arguments to see what arguments can be used. A minimal CLI call is for example:

> java -jar jplag-2.12.1-SNAPSHOT-jar-with-dependencies.jar your/sorcecode/folder -l c_cpp -s -r your/report/folder

Note that you seem to use the `target` directory as base code directory, which does not make a lot of sense. The base code is a common framework that is used in all submissions. For example, a stub in which students fill out an empty method body.

**Comment by m-jabari** [comment]
Thank you @tsaglam, JPlag (2.12.1) working fine without any issues.

**Comment by bloombar** [comment]
@tsaglam et al - I am new to JPlag and am having trouble running v`3.1.0` from the command line.  My process so far has been:
1. download repository
2. run `mvn clean package assembly:single` to generate a single `.jar` file with all dependencies.
3. run `java -jar jplag/target/jplag-3.1.0-SNAPSHOT-jar-with-dependencies.jar -l java -bc ~/Desktop/exam-1-given-code -S ~/Desktop/exam-1-submissions -r ~/Desktop/exam-1-reports`

The output says, `jplag: error: too few arguments`.  I have tried toying with the various other arguments, `-d`, `-p`, `-x`, `-t`, `-m`, `-n`, `-c`, but it's not clear which args are required and what values they expect.

Could someone please give me an example of a minimal v`3` command line execution, including which arguments are required and what their expected value ranges are?

**Reply by tsaglam** [reply]
In general, we recommend running a released version of JPlag, for example, v3.0.0 or the legacy version, as v3.1.0 (that will actually become 4.0.0) is still under development. Nevertheless, it seems you are using the subdirectory argument instead of the root directory argument. please try: `java -jar jplag/target/jplag-3.1.0-SNAPSHOT-jar-with-dependencies.jar -l java -bc ~/Desktop/exam-1-given-code -r ~/Desktop/exam-1-reports ~/Desktop/exam-1-submissions`

**Reply by tsaglam** [reply]
Also, note that Java is the default language so that means you can omit `-l java`.

**Reply by bloombar** [reply]
@tsaglam - many thanks.  This worked perfectly.

**Comment by m-jabari** [comment]
Maybe my snip code works for you,

sudo apt install openjdk-8-jre-headless   # version 8u252-b09-1ubuntu1
sudo apt install default-jre
sudo apt install maven

cd ~/Desktop
git clone https://github.com/jplag/jplag.git && cd jplag
mvn clean install

javac -version

#after generate 
java -jar jplag-3.0.0-SNAPSHOT-jar-with-dependencies.jar -h

java -jar jplag-2.12.1-SNAPSHOT-jar-with-dependencies.jar -l c/c++ -r ~/Desktop/report -v d ~/Desktop/fnFiles
javaws -J-Ddeployment.console.startup.mode=SHOW https://jplag.ipd.kit.edu/atujplag/launch.jnlp

**Reply by tsaglam** [reply]
Sorry, what is this referencing?
